src/services/node-service.ts

```diff
@@ -3,27 +3,35 @@ import { SignedTransactionInput } from "../graphql/input-types";
 
 const HTTP_PREFIX_PATTERN = /^http(s?):\/\//;
 
+export function checkNodeUrl(nodeUrl?: string): boolean {
+  const url = nodeUrl || process.env.ERGO_NODE_HOST || process.env.ERGO_NODE_ADDRESS;
+
+  if (!url) {
+    throw Error("ERGO_NODE_ADDRESS is undefined.");
+  } else if (!HTTP_PREFIX_PATTERN.test(url)) {
+    throw Error("ERGO_NODE_ADDRESS should be http:// or https:// prefixed.");
+  }
+
+  return true;
+}
+
 export class NodeService {
```

Comment:
I have a suggestion for this class,
We can use the `singleton` pattern for it since we only have one node service per instance and it shouldn't be instantiated more than once! What do you think?

Reply:
Yeah, makes sense. Will do

Reply:
Seems great now!